nilearn/_utils/fixes/sklearn_validation.py

```diff
@@ -0,0 +1,401 @@
+# scikit-learn v0.17
+# file: sklearn.utils.validation
+
+import numpy as np
+import warnings
+import scipy.sparse as sp
+
+
+class NotFittedError(ValueError, AttributeError):
+    """Exception class to raise if estimator is used before fitting
+    This class inherits from both ValueError and AttributeError to help with
+    exception handling and backward compatibility.
+    """
+
+
+def check_array(array, accept_sparse=None, dtype="numeric", order=None,
```

Comment:
> > +def check_array(array, accept_sparse=None, dtype="numeric", order=None,

> Can it be replaced by atleast2d_or_csr ? Is it the same or not ?

If the question is whether check_array is the same as atleast2d_or_csr,
the answer is no: it does additional checks and should be prefered in
general.